kong/global.lua

```diff
@@ -167,16 +167,43 @@ end
 function _GLOBAL.init_worker_events()
   -- Note: worker_events will not work correctly if required at the top of the file.
   --       It must be required right here, inside the init function
-  local worker_events = require "resty.worker.events"
+  local worker_events
+  local opts
 
-  local ok, err = worker_events.configure {
-    shm = "kong_process_events", -- defined by "lua_shared_dict"
-    timeout = 5,            -- life time of event data in shm
-    interval = 1,           -- poll interval (seconds)
+  local singletons = require "kong.singletons"
 
-    wait_interval = 0.010,  -- wait before retry fetching event data
-    wait_max = 0.5,         -- max wait time before discarding event
-  }
+  if singletons.configuration and
+     singletons.configuration.legacy_worker_events then
+
+    opts = {
+      shm = "kong_process_events", -- defined by "lua_shared_dict"
+      timeout = 5,            -- life time of event data in shm
+      interval = 1,           -- poll interval (seconds)
+
+      wait_interval = 0.010,  -- wait before retry fetching event data
+      wait_max = 0.5,         -- max wait time before discarding event
+    }
+
+    worker_events = require "resty.worker.events"
+
+  else
+    local sock_name = "worker_events.sock"
+    if ngx.config.subsystem == "stream" then
+      sock_name = "stream_" .. sock_name
+    end
+
+    opts = {
+      unique_timeout = 5,     -- life time of unique event data in lrucache
+      broker_id = 0,          -- broker server runs in nginx worker #0
+      listening = "unix:" ..  -- unix socket for broker listening
+                  ngx.config.prefix() .. sock_name,
+    }
+
+    --worker_events = require "resty.events"
+    worker_events = require "resty.events.compat"
```

Comment:
Even when we load the `compat` module, should we make Kong to use new APIs everywhere in our code base. I do think we want to get rid of `compat`, and also people use our sources as an example, we should use the new api everywhere and just have `compat` there for custom plugins (and perhaps in case we missed to change something)? (Edit: I see this PR is not currently using the lib that has a new API, so just a reminder then).

Reply:
Yes, the new API is a breaking change, we need to change many code in kong. Now I create this compatible layer only to try pass ci test.